### PR TITLE
Feat(event-cms): Create Form Objek & Create Component 

### PR DESCRIPTION
**Pull request by MNAlfiansyah**
## Changes 

- add Component Input Text
- add Component Inputarea
- Add Component Input File Single
- Add Component input File Multiple
- Create Form For Object
- Refactor Code Object Add Button

## Preview
image
 

# Evidence
title: Create Form Objek & Create Component
project: Jabar Super Apps
participants: @rayfajars @johndy2742 @ristianaditya @MNAlfiansyah

### CODE REVIEW

File: dashboard/components/Base/DragAndDropFile/index.vue [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+  const fileInputIsChange = ref(false);
+  const proggresBarIsSuccess = ref(false);
+  const percentageProggres = ref(0);
+  const intervalPercentage = ref(null);
+  const formatSizeFile = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
+  const responseImage = ref('');
+  const fileIsCorrect = ref(false);
+  const disabledButton = ref(true);
+  
+//   const router = useRouter();
```
Comment (rayfajars) [inline dashboard/components/Base/DragAndDropFile/index.vue]: is this code used?

File: dashboard/components/Base/DragAndDropFile/index.vue [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+      files.value = e.target.files[0];
+      dataFiles.value.name = files.value.name;
+      dataFiles.value.mimeType = files.value.type;
+      dataFiles.value.fileSize =  convertSize(files.value.size);
+      fileInputIsChange.value = true;
+      convertFileToBase64(files.value);
+      runProgressBar();
+      checkFileValidation();
+      dataFiles.value.fileCorrect = fileIsCorrect.value;
+      // const imageFile = { ...dataFiles.value };
```
Comment (rayfajars) [inline dashboard/components/Base/DragAndDropFile/index.vue:174]: same as this

File: dashboard/components/Base/DragAndDropFile/index.vue [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+const resetDataFile = () => {
+  percentageProggres.value = 0;
+  proggresBarIsSuccess.value = false;
+  fileInputIsChange.value = false;
+  files.value = '';
+  responseImage.value = '';
+  fileIsCorrect.value = false;
+  disabledButton.value = true;
+//   $emit('disabled-button', disabledButton.value);
```
Comment (rayfajars) [inline dashboard/components/Base/DragAndDropFile/index.vue]: same as this

File: dashboard/components/Base/DragAndDropFile/index.vue [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+  };
+  reader.readAsDataURL(FileObject);
+};
+
+const checkFileValidation = () => {
+  if (files.value) {
+    if (FileSizeIsCompatible() && FormatFileIsCompatible()) {
+      fileIsCorrect.value = true;
+      disabledButton.value = false;
+    //   $emit('disabled-button', disabledButton.value);
```
Comment (rayfajars) [inline dashboard/components/Base/DragAndDropFile/index.vue]: same as this

Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFile/index.vue]: sorry kang i fixed it on newest commit https://github.com/jabardigitalservice/event-management-system/pull/17/commits/a024c669b57f847ffb69f1e1797ac72c090410a3

File: dashboard/components/Base/DragAndDropFile/index.vue (hunk 3 above)
Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFile/index.vue]: sorry kang i fixed it on newest commit https://github.com/jabardigitalservice/event-management-system/pull/17/commits/a024c669b57f847ffb69f1e1797ac72c090410a3

File: dashboard/components/Base/DragAndDropFile/index.vue (hunk 2 above)
Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFile/index.vue]: sorry kang i fixed it on newest commit https://github.com/jabardigitalservice/event-management-system/pull/17/commits/a024c669b57f847ffb69f1e1797ac72c090410a3

File: dashboard/components/Base/DragAndDropFile/index.vue (hunk 1 above)
Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFile/index.vue]: sorry kang i fixed it on newest commit https://github.com/jabardigitalservice/event-management-system/pull/17/commits/a024c669b57f847ffb69f1e1797ac72c090410a3

File: dashboard/components/Base/TextareaInput/index.vue [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  import { useField } from 'vee-validate'
+
+  const props = defineProps({
+    type: {
+      type: String,
+      default: 'text',
+    },
+    value: {
+      type: String,
+      default: undefined,
```
Comment (rayfajars) [inline dashboard/components/Base/TextareaInput/index.vue:35]: my defined default with `undefined` on type string ?

File: dashboard/components/Base/ViewFileModal/index.vue [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+              color="gray"
+              variant="ghost"
+              icon="i-heroicons-x-mark-20-solid"
+              class="-my-1"
+              @click="handleClose"
+            />
+          </div>
+        </template>
+        <iframe
+          v-if="props.mimeType === 'application/pdf'"
```
Comment (rayfajars) [inline dashboard/components/Base/ViewFileModal/index.vue:22]: why not just write      `v-if="mimeType === 'application/pdf'"` ?

File: dashboard/components/TheHeader.vue [hunk 7]
```diff
@@ -20,6 +20,13 @@
 <script setup lang="ts">
   import { useActivePage } from '@/store/index'
   const activePage = useActivePage()
+  const pageTitle = ref('')
+
+  onMounted(() => {
+    const route = useRoute()
+    pageTitle.value = route.meta?.title ?? ''
```
Comment (rayfajars) [inline dashboard/components/TheHeader.vue:27]: why not write it outside onMounted ?

File: dashboard/components/TheHeader.vue [hunk 8]
```diff
@@ -20,6 +20,13 @@
...
+  })
+ 
 </script>
 
 <style scoped></style>
```
Comment (rayfajars) [inline dashboard/components/TheHeader.vue:32]: is this used ?

Comment (MNAlfiansyah) [reply dashboard/components/TheHeader.vue:26]: before it for fixing the problem from title, but now it fixed by @ristianaditya, i already deleted it kang.
[newest commit for remove unused code](https://github.com/jabardigitalservice/event-management-system/pull/17/commits/5281c5d52a9fd26891b078c8b18ef8c5bfa562f8)

File: dashboard/components/Base/DragAndDropFileMultiple/index.vue [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+    return props.detailDragAndDrop.formatTypeFile.includes(files.value.type)
+  }
+
+  const fileDocument = (file) => {
+    return file.data ? base64ToBlobUrl(file.data, file.mimeType) : ''
+  }
+
+  const imageClicked = (data: object, index: number) => {
+    // Component Sementara (blm selesai)
+    // console.log(data, index);
```
[the remark this answers is not in the document]
Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFileMultiple/index.vue:294]: its used for the next update component kang, so i decide to keep it

File: dashboard/components/Base/DragAndDropFile/index.vue [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+      {{ sublabel }}
+    </p>
+    <div class="mt-2 flex w-full items-center justify-center">
+      <div
+        v-if="fileInputIsChange"
+        :class="[
+          fileIsCorrect
+            ? 'border-green-300 bg-green-50'
+            : 'border-red-300 bg-red-50',
+          props.heightDragAndDrop,
```
[the remark this answers is not in the document]
Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFile/index.vue]: fixed it kang [commit]([597fd53](https://github.com/jabardigitalservice/event-management-system/pull/17/commits/597fd53e0d4870b33b221dfba64e3749420a9435)), thanks for the advice

File: dashboard/components/Base/DragAndDropFileMultiple/index.vue (hunk 9 above)
[the remark this answers is not in the document]
Comment (rayfajars) [reply dashboard/components/Base/DragAndDropFileMultiple/index.vue:294]: used comment

// TODO : 

to define if there will be an update in the future

[comment todo](https://medium.com/imdoneio/the-imdone-pitch-feedback-welcome-386430accf01)

File: dashboard/components/Base/DragAndDropFileMultiple/index.vue [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+              <span class="ml-1 text-gray-500">atau</span>
+              <span class="ml-1 text-green-600 underline">Pilih File</span>
+            </p>
+          </div>
+          <input
+            id="drag-and-drop-file-multiple"
+            ref="file"
+            type="file"
+            class="hidden"
+            :disabled="dataFilesMultiple.length == 0 ? false : true"
```
[the remark this answers is not in the document]
Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFileMultiple/index.vue]: fixed it kang [commit]([7a09cf1](https://github.com/jabardigitalservice/event-management-system/pull/17/commits/7a09cf167d38bee3a51f13b9ab7da753e0b07dc8)), thanks for the advice

File: dashboard/components/Base/TextareaInput/index.vue (hunk 5 above)
Comment (MNAlfiansyah) [reply dashboard/components/Base/TextareaInput/index.vue]: fixed it kang [commit](https://github.com/jabardigitalservice/event-management-system/pull/17/commits/e6a1e0604517183aee82194c31f2b348d704f7ab), thanks for the advice

File: dashboard/components/Base/ViewFileModal/index.vue (hunk 6 above)
Comment (MNAlfiansyah) [reply dashboard/components/Base/ViewFileModal/index.vue]: fixed it kang [commit](https://github.com/jabardigitalservice/event-management-system/pull/17/commits/0bf5669f913e59656cc744fa49d58e93d2c7dfc5), thanks for the advice

File: dashboard/components/Base/DragAndDropFileMultiple/index.vue (hunk 9 above)
[the remark this answers is not in the document]
Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFileMultiple/index.vue]: thanks for the advice kang, i update the comment [commit](https://github.com/jabardigitalservice/event-management-system/pull/17/commits/cdbd815ee00bbf123017c113c98c01d6508233e0)

File: dashboard/components/TheHeader.vue (hunk 7 above)
Comment (MNAlfiansyah) [reply dashboard/components/TheHeader.vue]: this one already fixed kang 🙏 [commit](https://github.com/jabardigitalservice/event-management-system/pull/17/commits/5281c5d52a9fd26891b078c8b18ef8c5bfa562f8)

**Review by Ibwedagama — APPROVED** [review]

File: dashboard/components/Base/DragAndDropFileMultiple/index.vue [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+    }
+  }
+
+  const FileSizeIsCompatible = () => {
+    return files.value.size <= props.detailDragAndDrop.maxSizeFile
+  }
+
+  const FormatFileIsCompatible = () => {
+    return props.detailDragAndDrop.formatTypeFile.includes(files.value.type)
+  }
```
[the remark this answers is not in the document]
Comment (MNAlfiansyah) [reply dashboard/components/Base/DragAndDropFileMultiple/index.vue]: oke kang, i updated it sorry for my mistakes and thank for the correction [commit refactor name format to camelcase](https://github.com/jabardigitalservice/event-management-system/pull/17/commits/c6c5481c3937196b5b25ee20b8de3024e83c80c9)

**Review by marsellavaleria19 — APPROVED** [review]

**Review by rayfajars — APPROVED** [review]

**Pull request merged**